### ISSUE TITLE
Wrong hack count calculation

### ISSUE THREAD

**Issue by SebastienForay** [filecc/cooldown-calculator-ingress#3]
Hi 👋,

It seems that the calculated hack count is wrong.
As of https://ingress.fandom.com/wiki/Multi-Hack,

```
Diminishing returns applies when using multiple Multi-Hacks :
the rarest Multi-Hack provides a full bonus, while all others only provide half, e.g. two Common Multi-Hacks add 6 burnout resistance instead of 8.
```

So when you set the calculator to:
Rare MultiHack
Rare MultiHack
VeryRare MultiHack

The output is as follow:
```
Cooldown: 180 seconds

Hacks before burnout: 22

Minimum time to burnout: 63 minute and 0 seconds
```

But total number of hacks must be 24.
Calcul is:

Base hacks: +4
VRMH hacks : +12
RMH hacks : 2* +4

Total : 4+12+4+4 = 24

-----

Hope it helps 🙂

**Comment by filecc** [comment]
Hi! 
thanks, I’ll take a look asap. 😊

**Comment by ynad** [comment]
> Hi wave,
> 
> It seems that the calculated hack count is wrong. As of https://ingress.fandom.com/wiki/Multi-Hack,
> 
> ```
> Diminishing returns applies when using multiple Multi-Hacks :
> the rarest Multi-Hack provides a full bonus, while all others only provide half, e.g. two Common Multi-Hacks add 6 burnout resistance instead of 8.
> ```
> 
> So when you set the calculator to: Rare MultiHack Rare MultiHack VeryRare MultiHack
> 
> The output is as follow:
> 
> ```
> Cooldown: 180 seconds
> 
> Hacks before burnout: 22
> 
> Minimum time to burnout: 63 minute and 0 seconds
> ```
> 
> But total number of hacks must be 24. Calcul is:
> 
> Base hacks: +4 VRMH hacks : +12 RMH hacks : 2* +4
> 
> Total : 4+12+4+4 = 24
> 
> Hope it helps slightly_smiling_face

Issue should be fixed now with commit b38659d18e74ac64cbcfdb30760e7e51367b0ab4 
The sorting of arrays storing mod values was broken.
Thanks for pointing it out ;)

**Comment by SebastienForay** [comment]
> Issue should be fixed now with commit b38659d18e74ac64cbcfdb30760e7e51367b0ab4 
> The sorting of arrays storing mod values was broken.
> Thanks for pointing it out ;) 

It is, thanks a lot 🙂👍
